### ISSUE TITLE
[BUG] Stagur Or Randomise Feed Updates

### ISSUE THREAD

**Issue by hbednar** [FreshRSS/FreshRSS#5557]
**Describe the bug**
When you have a lot of feeds sometimes when doing a full feed update you get a lot fail, though after updating them manually they work.

To fix this can you either update feeds in small batches instead of all at once or introduce a random delay between each.

**To Reproduce**
1.Either allow automatic updates or full manual
2. Some feeds fail
3. Update again manually

**Expected behavior**
No working feeds should fail to update when doing a full update.

**Screenshots**

**Environment information (please complete the following information):**
- Device: Mixed
- OS: Debian 12
- Browser: Mixed
- FreshRSS version:  1.21.1-dev
- Database version:
- PHP version: 8.2.7
- Installation type: VM

**Additional context**

**Comment by Alkarex** [comment]
Hello,
Are your errors from feeds hosted on the same server?
Currently, the eligible feeds are refreshed starting by the one that last has been refreshed, so the order is not always the same.
Furthermore, you can (and should) tune the refresh frequency of your different feeds to avoid refreshing unnecessarily.
How many feeds are you talking about? (I have 200 feeds in my personal instance, and 8000 on a work-related instance, and the refresh mechanism works fine)
Do you have some error messages from the failed feeds?

**Comment by hbednar** [comment]
I have a single server with two accounts i believe there is over 400 feeds between them, is there some where i can look to confirm this. I have tuned the default refresh frequency under Archiving and one or two individual feeds.

The error i am intermittently getting is `cURL error 6: Could not resolve host:`

**Comment by Alkarex** [comment]
`Could not resolve host` is a sign of DNS (lookup) problem (or Internet connectivity) independent of FreshRSS

**Comment by hbednar** [comment]
Yes, i believe it's down to the number of requests at once as they all have the same time in the logs when it happens.

I think they start to time out as they all seem to be going through at once, at least that is what FreshRSS logs show.

**Comment by hbednar** [comment]
This is just an example and there are a lot more but, i couldn't fit them in:

![2023-08-08](https://github.com/FreshRSS/FreshRSS/assets/71215137/cc6a5304-918b-4f90-b73a-f628d4a7695e)

**Comment by Frenzie** [comment]
Could not resolve host typically returns very quickly (even more so if it's the exact same host) so if that's all the same host you're seeing could not resolve, next, could not resolve, next, etc.

**Comment by hbednar** [comment]
Like i said before, i think its down to a lot of requests at once, as i never get issues when updating any of these hosts one at a time.

For context, there is a long list of `cURL error 6` all for different hosts and all with very close times.
This only happens sometimes i think when all feeds start to update together.

looking at the logs, it looks like one or two hundred requests all fail together or with in one second apart, as there are several pages with the same time and error.

**Comment by Alkarex** [comment]
Feeds are not updated all at once, but sequentially. I do not believe this is related to FreshRSS but more likely to a temporary network connectivity issue or DNS misconfiguration.
 
If you like, you can make a mini [FreshRSS extension](https://github.com/FreshRSS/Extensions) listening to the `simplepie_before_init` event and add a PHP delay with [`sleep()`](https://php.net/function.sleep).

E.g. using the model of https://github.com/FreshRSS/FreshRSS/tree/88026e1e751c5e00b9fb5fc8e9f268b196882582/lib/core-extensions/Tumblr-GDPR

```php
<?php

final class DelayFeedExtension extends Minz_Extension {
	public function init(): void {
		$this->registerHook('simplepie_before_init', [self::class, 'simplepie_before_init']);
	}

	public static function simplepie_before_init($simplePie, FreshRSS_Feed $feed): void {
		sleep(2);
	}
}
```
